src/ServerTelemetryChannel/SamplingTelemetryProcessor.cs

```diff
@@ -217,17 +200,86 @@ public void Process(ITelemetry item)
+        private SamplingTelemetryItemTypes PopulateSamplingFlagsFromTheInput(string input)
         {
-            var excludedTypesHashSetRef = this.excludedTypesHashSet;
-            var includedTypesHashSetRef = this.includedTypesHashSet;
+            SamplingTelemetryItemTypes samplingTypeFlags = SamplingTelemetryItemTypes.None;
 
-            if (excludedTypesHashSetRef.Count > 0 && excludedTypesHashSetRef.Contains(telemetryItemType))
+            if (!string.IsNullOrEmpty(input))
+            {
+                string[] splitList = input.Split(this.listSeparators, StringSplitOptions.RemoveEmptyEntries);
+                foreach (string item in splitList)
+                {
+                    if (this.allowedTypes.ContainsKey(item))
+                    {
+                        switch (item.ToUpperInvariant())
+                        {
+                            case RequestTelemetryName:
+                                samplingTypeFlags |= SamplingTelemetryItemTypes.Request;
+                                break;
+                            case DependencyTelemetryName:
+                                samplingTypeFlags |= SamplingTelemetryItemTypes.RemoteDependency;
+                                break;
+                            case ExceptionTelemetryName:
+                                samplingTypeFlags |= SamplingTelemetryItemTypes.Exception;
+                                break;
+                            case PageViewTelemetryName:
+                                samplingTypeFlags |= SamplingTelemetryItemTypes.PageView;
+                                break;
+                            case TraceTelemetryName:
+                                samplingTypeFlags |= SamplingTelemetryItemTypes.Message;
+                                break;
+                            case EventTelemetryName:
+                                samplingTypeFlags |= SamplingTelemetryItemTypes.Event;
+                                break;
+                        }
+                    }
+                }
+            }
+
+            return samplingTypeFlags;
+        }
+
+        private void HandlePossibleProactiveSampling(ITelemetry item, double currentSamplingPercentage, ISupportAdvancedSampling samplingSupportingTelemetry = null)
+        {
+            var advancedSamplingSupportingTelemetry = samplingSupportingTelemetry ?? item as ISupportAdvancedSampling;
+
+            if (advancedSamplingSupportingTelemetry != null)
+            {
+                if (advancedSamplingSupportingTelemetry.IsProactivelySampledOut)
+                {
+                    this.proactivelySampledOutCounters.AddItems(advancedSamplingSupportingTelemetry.ItemTypeFlag, Convert.ToInt64(100 / currentSamplingPercentage));
+
+                    if (TelemetryChannelEventSource.IsVerboseEnabled)
+                    {
+                        TelemetryChannelEventSource.Log.ItemSampledOut(item.ToString());
```

Comment:
Do a different log here to indicate item was sampled out because of Proactive sampling set. The decision to drop item was made elsewhere, and logs should hint this class did not make this decision.